### PR TITLE
Update RateGroupDriver to new constructor and configure() function

### PR DESCRIPTION
**Pull request by timcanham**
| | |
|:---|:---|
|**_Originating Project/Creator_**| JPL Infrastructure work |
|**_Affected Component_**| `Svc/RateGroupDriver` |
|**_Affected Architectures(s)_**| All |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**| y |
|**_Builds Without Errors (y/n)_**| y |
|**_Unit Tests Pass (y/n)_**| y |
|**_Documentation Included (y/n)_**| y |

---
## Change Description

The rate group divider array was moved from the constructor to a new `configure()` function.

## Rationale

This makes the constructor compliant with the FPP code generator for constructors, simplifying the FPP statements

## Testing/Review Recommendations

Re-run unit tests and verify Ref app still runs.

## Future Work

None.

### CODE REVIEW

**Review by LeStarch — APPROVED** [review]
Looks like what I'd expect.  Is there a reason for the "Draft" status?

**Comment by timcanham** [comment]
I make them draft so I can do a final diff in the browser...

File: RPI/Top/instances.fpp [hunk 1]
```diff
@@ -313,18 +313,13 @@ module RPI {
   instance rateGroupDriverComp: Svc.RateGroupDriver base id 1700 \
   {
 
-    phase Fpp.ToCpp.Phases.configObjects """
+    phase Fpp.ToCpp.Phases.configComponents """
```
Comment (bocchino) [inline RPI/Top/instances.fpp:316]: I don't think we want to move the array declaration from the `configObjects` phase to the `configComponents` phase.

* The `configObjects` phase is for declaring static objects and has program-lifetime scope
* The `configComponents` phase is inside a function and has local function scope

Maybe it's OK to declare the array inside the function, if the array values get copied by the call, but this feels like a change we don't want to do.

**Review by bocchino — CHANGES_REQUESTED** [review]
Looks good, but please review the comment.

**Review by bocchino — APPROVED** [review]
Looks good!

**Comment by timcanham** [comment]
The 'configure()' function actually dies copy it, but it makes sense to keep it consistent with 'ActiveRateGroup'. Still needs to be tested, so don't merge it yet.

**Comment by timcanham** [comment]
OK - Linux testing done.

**Pull request merged**